### PR TITLE
Primitive standard added

### PR DESCRIPTION
**Pull request by psavery**
The 'Added "Reduce (Primitive and Standardize)"' is the important commit here. Just look at that commit if you want to see the diff. It is dependent upon updating spglib, which was sent in as a pull request here: https://github.com/cryos/avogadro/pull/838

### CODE REVIEW

**Comment by psavery** [comment]
This will subsequently be used in conjunction with Yaehmop. Look at the commit description for 'Added "Reduce (Primitive and Standardize)"' for  more details of what it does.

**Comment by psavery** [comment]
I just updated this one as well with the spglib 1.9.7 update. It seems to work just fine.

File: libavogadro/src/extensions/crystallography/crystallographyextension.cpp [hunk 1]
```diff
@@ -2191,6 +2195,13 @@ namespace Avogadro
     CE_CACTION_DEBUG(PrimitiveReduceIndex);
     CE_CACTION_ASSERT(PrimitiveReduceIndex);
 
+    // PrimitiveReduceStandardIndex,
+    a = new QAction(tr("Reduce Cell (&Primitive and Standardize)"), this);
```
Comment (ghutchis) [inline libavogadro/src/extensions/crystallography/crystallographyextension.cpp]: The rest of the patch looks fine - can we pick a better name, like "Reduce and Standardize"?

Comment (psavery) [inline libavogadro/src/extensions/crystallography/crystallographyextension.cpp]: Yeah we can. Technically, there is a "Niggli reduction" as well, so to make sure don't confuse the two types of reduction, I think we need the word "Primitive"

Is "Primitive Reduce and Standardize" too long? Or will that work?

Comment (ghutchis) [inline libavogadro/src/extensions/crystallography/crystallographyextension.cpp]: Sorry - getting back to this. I think that's probably fine.

**Comment by psavery** [comment]
It is now updated with the changes suggested.

**Pull request merged**